template/pdk.gen.go.ejs

```diff
@@ -14,26 +14,24 @@ func _<%- goName(ex.name) %>() int32 {
 	var err error
 	_ = err
   <% if (ex.input) { -%>
-    <% if (ex.input.contentType === 'application/json') { -%>
+    <% if (isJsonEncoded(ex.input)) { -%>
 			pdk.Log(pdk.LogDebug, "<%- goName(ex.name) %>: getting JSON input")
 			var input <%- toGolangType(ex.input) %>
 			err = pdk.InputJSON(&input)
 			if err != nil {
 				pdk.SetError(err)
 				return -1
 			}
-    <% } else if (ex.input.type === 'string') { -%>
-      <% if (ex.input.$ref && ex.input.$ref.enum) { -%>
+    <% } else if (isString(ex.input)) { -%>
+			pdk.Log(pdk.LogDebug, "<%- goName(ex.name) %>: getting string input")
+			input := pdk.InputString()
+    <% } else if (isEnum(ex.input)) { -%>
 			pdk.Log(pdk.LogDebug, "<%- goName(ex.name) %>: getting enum string input")
-      input, err := stringTo<%- goName(ex.input.$ref.name) %>(pdk.InputString())
+      input, err := stringTo<%- goName(ex.input.xtpType.name) %>(pdk.InputString())
```

Comment:
I might just want to refactor goName to take an `XtpTyped` object. so you can just do `goName(ex.input)` or any typed object works as a parameter

Reply:
sgtm! I can't think of something we wouldn't be able to handle inside the function vs. passing the specific value in  e.g. `ex.input` vs. `ex.input.xtpType.name`